packages/playground/blueprints/src/lib/compile.ts

```diff
@@ -71,6 +92,14 @@ export function compileBlueprint(
 		progress = new ProgressTracker(),
 		semaphore = new Semaphore({ concurrency: 3 }),
 		onStepCompleted = () => {},
+		onRunningSteps = (progressMap: PromiseMap) => {
+			for (const promise of progressMap.values()) {
+				promise.catch((error) => {
+					throw error;
+				});
```

Comment:
is this necessary? won't it already raise an error if we don't handle the catch in an `async`/`await` block?

Reply:
Removed.